astropy/modeling/functional_models.py

```diff
@@ -1054,49 +1062,84 @@ class Voigt1D(Fittable1DModel):
     fwhm_L = Parameter(default=2/np.pi)
     fwhm_G = Parameter(default=np.log(2))
 
-    _abcd = np.array([
-        [-1.2150, -1.3509, -1.2150, -1.3509],  # A
-        [1.2359, 0.3786, -1.2359, -0.3786],    # B
-        [-0.3085, 0.5906, -0.3085, 0.5906],    # C
-        [0.0210, -1.1858, -0.0210, 1.1858]])   # D
+    sqrt_pi = np.sqrt(np.pi)
+    sqrt_ln2 = np.sqrt(np.log(2))
+    sqrt_ln2pi = np.sqrt(np.log(2) * np.pi)
+    _last_z = np.zeros(1, dtype=complex)
+    _last_w = np.zeros(1, dtype=float)
+    _wofz = None
+    _last_wofz = None
+
+    def __new__(cls, x_0=x_0.default, amplitude_L=amplitude_L.default, fwhm_L=fwhm_L.default,
+                fwhm_G=fwhm_G.default, method='hum2zpf16c', n_models=None, **kwargs):
+        if str(method).lower() in ('wofz', 'faddeeva', 'scipy'):
+            try:
+                from scipy.special import wofz
+            except (ValueError, ImportError) as err:
+                raise ImportError(f'Voigt1D method {method} requires scipy: {err}.') from err
+            cls._wofz = wofz
+        elif str(method).lower() in ('humlicek2', 'hum2zpf16c'):
+            cls._wofz = cls.hum2zpf16c
+        else:
+            raise ValueError(f'Not a valid method for Voigt1D Faddeeva function: {method}.')
```

Comment:
Should this just be:
```suggestion
            raise ValueError(f'Not a valid method for Voigt1D function: {method}.')
```
`Faddeeva` is only one of the approximations, right?

Reply:
It rather is a mathematical term for the complex error function used here; in principle unrelated to the specific approximation (so Faddeeva, wofz, errfc and so on are used interchangeably a lot, which I should perhaps have avoided...)

Reply:
I have left it at the function name for now, since it is not (intended to be) public anyway; though perhaps then the function should become `_faddeeva`, or `_erfcx_wrapper`...

Reply:
OK, I think sticking to "Humlicek2" for the name should not be too much of a burden, as it is the default and people will rarely have to type it explicitly. Have to decide then whether "scipy" or "wofz" is more meaningful for the alternative.

Reply:
I take it that `wofz` for Scipy's Faddeeva function is accepted; it is consistent in that both `method` options are the respective function names; and the `scipy.special` context should be adequately documented.